spark3/src/main/java/org/apache/iceberg/spark/actions/BaseTableMigrationSparkAction.java

```diff
@@ -99,20 +93,17 @@
     this.sourceTableLocation = CatalogUtils.URIToString(sourceCatalogTable.storage().locationUri().get());
   }
 
-  @Override
-  public CreateAction withProperties(Map<String, String> properties) {
-    this.additionalProperties.putAll(properties);
-    return this;
+  protected void setDestCatalogAndIdent(CatalogPlugin catalog, Identifier ident) {
```

Comment:
Called from `as`.

Reply:
We let you use "as" with Migration? Maybe I'm forgetting how this worked.

Reply:
I think this was needed before since Snapshot and Migrate descended from CreateTable Actions, now I don't think you need dest catalog at all

Reply:
This is still the parent class where we need to reference the dest catalog. However, I think your idea makes sense. We can probably remove the dest fields from the parent class as they are initialized differently now. Then we don't have to do that weird if statement in the constructor.

I've pushed an update. Let me know if you prefer the old approach.